src/framework/mpas_framework.F

```diff
@@ -85,22 +85,31 @@ subroutine mpas_framework_init_phase2(domain, io_system, calendar)!{{{
       call mpas_pool_set_error_level(MPAS_POOL_WARN)
 #endif
 
-      call mpas_pool_get_config(domain % configs, 'config_calendar_type', config_calendar_type)
-      call mpas_pool_get_config(domain % configs, 'config_pio_num_iotasks', config_pio_num_iotasks)
-      call mpas_pool_get_config(domain % configs, 'config_pio_stride', config_pio_stride)
-
       if (present(calendar)) then
          call mpas_timekeeping_init(calendar)
       else
+         call mpas_pool_get_config(domain % configs, 'config_calendar_type', config_calendar_type)
          call mpas_timekeeping_init(config_calendar_type)
       end if
 
-      pio_num_iotasks = config_pio_num_iotasks
-      pio_stride = config_pio_stride
-      if (pio_num_iotasks == 0) then
-         pio_num_iotasks = domain % dminfo % nprocs
+      !
+      ! Note: pio_num_iotasks and pio_stride are only used in MPAS_io_init if io_system is
+      !       not present. In stand-alone configurations, we expect that io_system will not
+      !       be present and that pio_num_iotasks and pio_stride will be available from
+      !       the namelist; in other systems, a PIO io_system may be provided.
+      !
+      if (.not. present(io_system)) then
+         call mpas_pool_get_config(domain % configs, 'config_pio_num_iotasks', config_pio_num_iotasks)
+         call mpas_pool_get_config(domain % configs, 'config_pio_stride', config_pio_stride)
+         pio_num_iotasks = config_pio_num_iotasks
+         pio_stride = config_pio_stride
+         if (pio_num_iotasks == 0) then
+            pio_num_iotasks = domain % dminfo % nprocs
+         end if
       end if
+
       domain % ioContext % dminfo => domain % dminfo
+
       call MPAS_io_init(domain % ioContext, pio_num_iotasks, pio_stride, io_system)
```

Comment:
For clarity, it would be good to initialize `pio_num_iotasks` and `pio_stride` to something before the call to `MPAS_io_init` with an `else` clause to the if-construct added above.  I realize that if `io_system` is provided then those two variables are unused in `MPAS_io_init`, but that's not obvious in this routine. and it still seems potentially dangerous (if that changes in the future), or at least confusing, to leave them uninitialized.

Reply:
That's a good point -- that we should never pass uninitialized values to non-optional subroutine arguments -- though I would probably disagree that what we're doing is confusing given the [prominent comment block](https://github.com/MPAS-Dev/MPAS-Model/blob/aa5d8d92c2a5a5cc85c0f7a63a967b8f803be7c8/src/framework/mpas_framework.F#L95-L100) explaining what is happening. I'll push another commit adding an else clause that initializes `pionum_iotasks` and `pio_stride`.

Reply:
@mgduda , thanks for making my suggestion.  It looks like the else clause initializes them to unassigned pointers now, though.  How about 0 or -1 or something?

Reply:
Ugh -- that's really bad of me! I had copy-pasted the assignment from the `if` case to the `else` case, got momentarily distracted, and forgot to actually clean up the assignments before committing and pushing. What do you think about setting `pio_stride` to 1 and `pio_num_iotasks` to 0 (so that after the if-else block, `pio_num_iotasks` would be set to the total number of MPI tasks)? This would correspond to what I believe is the default in all core's `Registry.xml` files.

Reply:
Ok, I thought it might be something like that. :)  I like the idea of something like -1 or -999 to hint that these values are not actually going to be used.  But I'm fine with whatever you prefer, I was primarily interested in seeing some initialized value in there.

Reply:
Using `-1` sounds reasonable to me. I'll also add line comments to state that these values are never expected to be used when an external `io_system` is provided. I should be able to push an amended commit in about 30 minutes.

Reply:
@mgduda , thanks for dealing with my picky requests. :)  Looks good.